hummingbot/connector/exchange/btc_markets/btc_markets_exchange.py

```diff
@@ -0,0 +1,966 @@
+            raise Exception("cancel_all can only be used when trading_pairs are specified.")
+        tracked_orders: Dict[str, BtcMarketsInFlightOrder] = self._in_flight_orders.copy().items()
+        cancellation_results = []
+        try:
+            tasks = []
+
+            for _, order in tracked_orders:
+                api_params = {}
+                if not order.is_cancelled:
+                    tasks.append(self._api_request(method="DELETE",
+                                                   path_url=f"{Constants.ORDERS_URL}/{order.exchange_order_id}",
+                                                   params=api_params,
+                                                   is_auth_required=True))
+
+            await safe_gather(*tasks)
+
+            open_orders = await self.get_open_orders()
+            for cl_order_id, tracked_order in tracked_orders:
+                open_order = [o for o in open_orders if o.client_order_id == cl_order_id]
+                if not open_order:
+                    cancellation_results.append(CancellationResult(cl_order_id, True))
+                    self.trigger_event(MarketEvent.OrderCancelled,
+                                       OrderCancelledEvent(self.current_timestamp, cl_order_id))
+                    self.stop_tracking_order(cl_order_id)
+                else:
+                    cancellation_results.append(CancellationResult(cl_order_id, False))
+        except Exception:
+            self.logger().network(
+                "Failed to cancel all orders.",
+                exc_info=True,
+                app_warning_msg="Failed to cancel all orders on BTCMarkets.net. Check API key and network connection."
+            )
+        return cancellation_results
+
+    def tick(self, timestamp: float):
+        """
+        Is called automatically by the clock for each clock's tick (1 second by default).
+        It checks if status polling task is due for execution.
+        """
+        now = time.time()
+        poll_interval = (self.SHORT_POLL_INTERVAL
+                         if now - self._user_stream_tracker.last_recv_time > 60.0
+                         else self.LONG_POLL_INTERVAL)
+        last_tick = int(self._last_timestamp / poll_interval)
+        current_tick = int(timestamp / poll_interval)
+        if current_tick > last_tick:
+            if not self._poll_notifier.is_set():
+                self._poll_notifier.set()
+        self._last_timestamp = timestamp
+
+    def get_fee(self,
+                base_currency: str,
+                quote_currency: str,
+                order_type: OrderType,
+                order_side: TradeType,
+                amount: Decimal,
+                price: Decimal = s_decimal_NaN,
+                is_maker: Optional[bool] = None) -> AddedToCostTradeFee:
+
+        safe_ensure_future(self.async_get_fee(base_currency, quote_currency, order_type, order_side, amount,
+                                              price, is_maker))
```

Comment:
What is the intention when calling `async_get_fee`? It seems like it is not required.

Reply:
As per https://www.btcmarkets.net/fees - "Trading fees are based on a rolling 30-day total of AUD trading volume, across all AUD trades. Trading volume is calculated every hour. The relevant trading fee is applied when an order is created.". 

Agree async_get_fee is probably in the wrong place and needed to be refactored into TradeFeeSchema (I think) and refreshed ~hourly.